### PR TITLE
add reference aggregates

### LINKED ISSUE #473: add default aggregates
Probably a python module and a json file in `io/reference_observations`

### PR DESCRIPTION
**Pull request by alorenzo175**
- [x] Closes #473 .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [ ] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: solarforecastarbiter/io/reference_aggregates.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                )
+            )
+    agg_dict = agg_def.copy()
+    agg_dict['observations'] = tuple(agg_obs)
+    agg = datamodel.Aggregate(**agg_dict)
+    return agg
+
+
+def make_reference_aggregates(token, provider, base_url,
+                              aggregates=REF_AGGREGATES):
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_aggregates.py:206]: do we care about using a mutable default value?

File: solarforecastarbiter/io/reference_aggregates.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                'from': '2019-01-01T08:00Z',
+                'until': None,
+            },
+        ]
+    }
+]
+
+
+def generate_aggregate(observations, agg_def):
+    agg_obs = []
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_aggregates.py:167]:
```suggestion
    """Generate an aggregate object.
    
    Parameters
    ----------
    observations: list of datamodel.Observation
    agg_def: dict
        Text metadata to create a datamodel.Aggregate.
        'observation' field names will be matched against names
        of datamodel.Observation in ``observations``.
    
    Returns
    -------
    datamodel.Aggregate
    """
    agg_obs = []
```

Comment (wholmgren) [reply solarforecastarbiter/io/reference_aggregates.py:167]: not sure if this should be a comment or a docstring, also not sure how to put this concisely, but something about replacing text description of observations in ``agg_def`` with observation objects obtained from the API

File: solarforecastarbiter/io/reference_observations/srml_reference_sites.json [hunk 3]
```diff
@@ -307,7 +307,7 @@
                 "site_id": "",
                 "timezone": "Etc/GMT+8"
             },
-            "uncertainty": 0,
+            "uncertainty": null,
```
Comment (wholmgren) [inline solarforecastarbiter/io/reference_observations/srml_reference_sites.json:310]: do we need a database migration?

File: solarforecastarbiter/io/tests/test_reference_aggregates.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import pandas as pd
+import pytest
+from solarforecastarbiter.io import reference_aggregates
+
+
+AGGDEF = {
```
Comment (wholmgren) [inline solarforecastarbiter/io/tests/test_reference_aggregates.py:6]: I would have thought you'd want a fixture that returns this dict.

**Review by wholmgren — APPROVED** [review]
Very cool!

**Comment by wholmgren** [comment]
Oh and add a brief "Reference aggregates" section after [Reference observations](https://solarforecastarbiter-core.readthedocs.io/en/latest/api.html#reference-observations)

File: solarforecastarbiter/io/reference_observations/srml_reference_sites.json (hunk 3 above)
Comment (alorenzo175) [reply solarforecastarbiter/io/reference_observations/srml_reference_sites.json:310]: good question. This was just to make that one observation consistent with all the rest, but I think they all currently have uncertainty==0 in prod right now, as do basically all of the other reference observations. We probably don't want them to be null but need to think about what a reasonable number for at least each network/variable

**Pull request merged**